src/pages/iou/withMoneyRequest.js

```diff
@@ -0,0 +1,116 @@
+import React, {createContext, forwardRef, useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import _ from 'underscore';
+import getComponentDisplayName from '../../libs/getComponentDisplayName';
+import withCurrentUserPersonalDetails, {withCurrentUserPersonalDetailsDefaultProps, withCurrentUserPersonalDetailsPropTypes} from '../../components/withCurrentUserPersonalDetails';
+import * as PersonalDetails from '../../libs/actions/PersonalDetails';
+import optionPropTypes from '../../components/optionPropTypes';
+import Navigation from '../../libs/Navigation/Navigation';
+import ROUTES from '../../ROUTES';
+import useOnNetworkReconnect from '../../components/hooks/useOnNetworkReconnect';
+
+const MoneyRequestContext = createContext();
+const moneyRequestPropTypes = PropTypes.shape({
+    // Money request amount
+    amount: PropTypes.number.isRequired,
+
+    // Money request currency
+    currency: PropTypes.string.isRequired,
+
+    // Selected participants from MoneyRequestParticipantsPage with login
+    participants: PropTypes.arrayOf(optionPropTypes),
+
+    // Money request description
+    comment: PropTypes.string.isRequired,
+
+    // Callback to update money request amount
+    setAmount: PropTypes.func.isRequired,
+
+    // Callback to update money request currency
+    setCurrency: PropTypes.func.isRequired,
+
+    // Callback to update money request participants
+    setParticipants: PropTypes.func.isRequired,
+
+    // Callback to update money request description
+    setComment: PropTypes.func.isRequired,
+
+    // Helper function to prevent user going further steps without completeing the previous steps
+    redirectIfEmpty: PropTypes.func.isRequired,
+});
+
+/**
+ * Redirect to money request initial page if one of the data is empty
+ * @param {Array} data
+ * @param {String} iouType
+ * @param {String} reportID
+ */
+// eslint-disable-next-line rulesdir/prefer-early-return
+const redirectIfEmpty = (data, iouType, reportID) => {
+    // If one of the data is empty, redirect the user to the initial money request page.
+    // If the data is a number (amount), 0 will be treated as empty.
+    if (_.some(data, (d) => (_.isNumber(d) ? d === 0 : _.isEmpty(d)))) {
+        Navigation.goBack();
+        Navigation.navigate(ROUTES.getMoneyRequestRoute(iouType, reportID));
+    }
+};
```

Comment:
Yes, I think it's better to check the `data` on each corresponding page.

Reply:
Done

Reply:
Why we are navigating back before navigating to the correct route?

Reply:
To close the current page

Reply:
The alternative is to introduce a way to replace current routes, so we don't need to go back anymore.

`Navigation.navigate(ROUTES.getMoneyRequestRoute(iouType.current, reportID.current), 'REPLACE');`

Then update `linkTo` to support navigation replace action.

```diff
diff --git a/src/libs/Navigation/linkTo.js b/src/libs/Navigation/linkTo.js
index 6f293ff057..8fb510ec70 100644
--- a/src/libs/Navigation/linkTo.js
+++ b/src/libs/Navigation/linkTo.js
@@ -1,7 +1,7 @@
 import {getStateFromPath, getActionFromState} from '@react-navigation/core';
 import linkingConfig from './linkingConfig';
 
-export default function linkTo(navigation, path) {
+export default function linkTo(navigation, path, type = 'NAVIGATE') {
     const normalizedPath = !path.startsWith('/') ? `/${path}` : path;
     if (navigation === undefined) {
         throw new Error("Couldn't find a navigation object. Is your component inside a screen in a navigator?");
@@ -22,7 +22,10 @@ export default function linkTo(navigation, path) {
         root = current;
     }
 
-    const action = getActionFromState(state, linkingConfig.config);
+    const action = {
+        ...getActionFromState(state, linkingConfig.config),
+        type,
+    };
 
     if (action !== undefined) {
         root.dispatch(action);
```

But I guess we don't want to do this because we are rebooting the navigation.

Reply:
What problem is this solving? i.e. what would go wrong if we simply navigate?

Reply:
Let say I directly access /send/new/confirmation. If we just simply navigate, we can go back and arrive at confirmation page.

https://github.com/Expensify/App/assets/50919443/a15242d7-bbef-4ded-a306-ff54a83892bf